pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,431 @@
+
+    def iter_names(self):
+        """
+        Overrides the abstract method.
+
+        Returns iterator for names of the objects in the object store
+
+        """
+        return six.iterkeys(self._data)
+
+    def iter_values(self, copy=True):
+        """
+        Overrides the abstract method.
+
+        Returns an iterator for the object values in the object store
+        """
+        for value in six.itervalues(self._data):
+            if copy:
+                yield(value.copy())
+            else:
+                yield(value)
+
+    def len(self):
+        """
+        Overrides the abstract method.
+
+        :class:`~pywbem_mock._baserepository.ObjectStoreAPI.len`
+
+        Returns count of number of objects in the object store dictionary.
+        """
+        return len(self._data)
+
+
+class InMemoryRepository(BaseRepositoryAPI):
+    """
+    Defines the data store and access methods for a simple CIMClass,
+    CIMInstance, and CIMQualifierDeclaration repository that maintains the date
+    in memory.
+
+    The API for this data store is defined in the BaseRepositoryAPI class and
+    the ObjectStoreAPI class.
+    """
+    def __init__(self, initial_namespace=None):
+        """
+        Initialize the InMemoryRepository by creating the top level dictionary.
+        The initial_namespace is optional.  If provided, that namespace
+        is added to the empty repository.
+
+        However, if there is a persistence mechanism, that would define
+        the namespaces from the persistent data so the initial namespace would
+        not be used.
+
+          Parameters:
+             initial_namespace:(:term:`string` or None):
+                Optional initial namespace to be defined for this repository.
+        """
+
+        # Create the in memory repository.  This defines the top level
+        # NocaseDict which defines the namespaces.
+        self._repository = NocaseDict()
+
+        # If an initial namespace is defined, add it to the repository
+        if initial_namespace:
+            self.add_namespace(initial_namespace)
+
+    def print_repository(self, dest=None, ):
+        """
+        Display the items in the repository. This displays information on
+        the items in the data base and is only a diagnostic tool.
+
+        Parameters:
+          dest (:term:`string`):
+            File path of the output file. If `None`, the output is written to
+            stdout.
+        """
+        def objstore_info(objstore_name):
+            """
+            Display the date for the object
+            """
+            for ns in self._repository:
+                if objstore_name == 'class':
+                    repo = self.get_class_repo(ns)
+                elif objstore_name == 'qualifier':
+                    repo = self.get_qualifier_repo(ns)
+                elif objstore_name == 'instance':
+                    repo = self.get_instance_repo(ns)
+                else:
+                    assert False
```

Comment:
Since this is a case that cannot be tested without modifying the internal data of the object, this will likely never be tested. To avoid untested code, I suggest to move the assert into the last else, like this:
```
                else:
                    assert objstore_name == 'instance'
                    repo = self.get_instance_repo(ns)
```

Reply:
OK. DONE